### DISCUSSION TITLE
How can I get a list with the indices of my key-value array?

### DISCUSSION THREAD

**Discussion by dabl03** [open-source-parsers/jsoncpp#1474] [Q&A] [answered]
I don't know, but I would like a more abundant documentation than the one offered by this repository generated by doxygen.

I have this json:
map.json:
{
    //The name is not important, I would like to give this freedom to modify the user of the game
    "world_1":{
        ...
    },
    //But that means I didn't know the name of the worlds
    "world_2":{
        ...
    }
}
main.cpp
json::Value root;
get_world_json("map",&root);
//But now I don't know what the worlds are called, so I can't do:
map.add_map(root["world_1"]);
//What I would like to do is something javascript style that you get the indices of your key-value array:
js:
var arr={"world_1":{...},"world_2":{...},...};
for (let index of arr)
    console.log("key: "+index+" is value: "+arr[index]);

**Comment by BillyDonahue** [comment, marked as answer]
Value has member functions to get you a list of keys. Value is also a C++ range, having begin and end member functions.

I think the library already does what you want. You just have to say it in C++ and not JS.

```c++
Value arr{
  {"world_1",{...}},
  {"world_2",{...}},
  ...
};

for (auto&& [k,v] : arr) {
  std::cout
    << "  " << k << ": "
    << v << std::endl;
}
```

this is very idiomatic C++ container iteration. I think you have to come with some level of experience to use the library. This could be making life a little harder for people don't have or need that basic familiarity to interpret the existing robo-docs.

There could be very basic examples like this added to the README.md or something.

**Reply by dabl03** [reply]
> Value tiene funciones miembro para obtener una lista de claves. El valor también es un rango de C++, que tiene funciones de miembro inicial y final.
> 
> Creo que la biblioteca ya se opone a lo que quieres. Solo tienes que decirlo en C++ y no en JS.

I don't know how to express the idea I have without mentioning a high level language. I am new to using json in c++. What do you recommend me?
